internal/core/services/events_forwarder/events_forwarder_service.go

```diff
@@ -24,60 +24,169 @@ package events_forwarder
+					return err
+				}
+				err = es.forwardRoomEvent(ctx, event, eventType, *instance, *scheduler, _forwarder)
+				if err != nil {
+					return err
+				}
+			case events.PlayerEvent:
+				err = es.forwardPlayerEvent(ctx, event, eventType, _forwarder)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	} else {
+		es.logger.Info(fmt.Sprintf("scheduler \"%v\" do not have forwarders configured", event.SchedulerID))
+	}
+
+	return nil
 }
 
-func (es *EventsForwarderService) forwardRoomEvent(ctx context.Context, room *game_room.GameRoom, instance *game_room.Instance, attributes map[string]interface{}, options interface{}) error {
-	err := es.eventsForwarder.ForwardRoomEventObsolete(ctx, room, instance, attributes, options)
+func (es *EventsForwarderService) forwardRoomEvent(
+	ctx context.Context,
+	event *events.Event,
+	eventType string,
+	instance game_room.Instance,
+	scheduler entities.Scheduler,
+	_forwarder *forwarder.Forwarder,
+) error {
+	var selectedPort int32
+	if len(instance.Address.Ports) > 0 {
+		selectedPort = instance.Address.Ports[0].Port
+	} else {
+		return errors.New("no room port found to forward roomEvent")
+	}
+
+	roomEvent := events.RoomEventType(eventType)
+
+	var pingType events.RoomPingEventType
+	if roomEvent == events.Ping {
+		if _type, ok := event.Attributes["pingType"]; ok {
+			pingType = events.RoomPingEventType(_type.(string))
+		} else {
+			return errors.New("roomEvent of type ping must contain key \"pingType\" in eventAttributes")
+		}
+
+	}
+
+	roomAttributes := events.RoomEventAttributes{
+		Game:      scheduler.Game,
+		RoomId:    event.RoomID,
+		Host:      instance.Address.Host,
+		Port:      strconv.Itoa(int(selectedPort)),
+		EventType: roomEvent,
+		PingType:  &pingType,
+		Other:     event.Attributes,
+	}
+	err := es.eventsForwarder.ForwardRoomEvent(ctx, roomAttributes, *_forwarder)
 	if err != nil {
-		reportRoomEventForwardingFailed(room.SchedulerID)
-		es.logger.Error(fmt.Sprintf("Failed to forward room events for room %s and scheduler %s", room.ID, room.SchedulerID), zap.Error(err))
+		reportRoomEventForwardingFailed(event.SchedulerID)
+		es.logger.Error(fmt.Sprintf("Failed to forward room events for room %s and scheduler %s", event.RoomID, event.SchedulerID), zap.Error(err))
 		return err
 	}
-	reportRoomEventForwardingSuccess(room.SchedulerID)
+
+	reportRoomEventForwardingSuccess(event.SchedulerID)
 	return nil
 }
 
-func (es *EventsForwarderService) forwardPlayerEvent(ctx context.Context, room *game_room.GameRoom, attributes map[string]interface{}, options interface{}) error {
-	err := es.eventsForwarder.ForwardPlayerEventObsolete(ctx, room, attributes, options)
+func (es *EventsForwarderService) forwardPlayerEvent(
+	ctx context.Context,
+	event *events.Event,
+	eventType string,
+	_forwarder *forwarder.Forwarder,
+) error {
+	var playerId string
```

Comment:
@caiordjesus here you could change if declarations to avoid statement like previous comments